students/preactive/Lesson3/list_lab.py

```diff
@@ -0,0 +1,142 @@
+    # Display the list (plain old print() is fine…).
+    fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+    print(fruits)
+
+    # Ask the user for another fruit and add it to the end of the list.
+    # Display the list.
+    response_one = input("Add a fruit:\n>")
+    fruits.append(response_one)
+    print(fruits)
+
+    # Ask the user for a number and display the number back to the user and
+    # the fruit corresponding to that number (on a 1-is-first basis).
+    # Remember that Python uses zero-based indexing,
+    # so you will need to correct.
+    num_fruits = len(fruits)
+    response_two_txt = "Pick a integer from 1 to %s:\n>" % str(num_fruits)
+    response_two = input(response_two_txt)
+    try:
+        response_two_int = int(response_two)
+        if response_two_int not in range(1, num_fruits + 1):
+            print("Not in range. Try again, mouth breather.")
+            series_one()
+        else:
+            print(response_two + ": " + str(fruits[response_two_int - 1]))
+    except:
+        print("Not in ranges. Try again, mouth breather.")
+        series_one()
+
+    # Add another fruit to the beginning of the
+    # list using “+” and display the list.
+    fruits = ["tomato"] + fruits
+    print(fruits)
+
+    # Add another fruit to the beginning of the list
+    # using insert() and display the list.
+    fruits.insert(len(fruits), "banana")
+    print(fruits)
+
+    # Display all the fruits that begin with “P”, using a for loop.
+    for fruit in fruits:
+        try:
+            if fruit[0:1].lower() == "p":
+                print(fruit)
+        except:
+            print("Something went wrong with lower method. "
+                  "But did you know you could save 15% or more "
+                  "on your car insurance by switching to Geiko?")
+    return fruits
+
+
+def series_two():
+    series_one_stub = series_one()
+
+    # Display the list.
+    print(series_one_stub)
+
+    # Remove the last fruit from the list.
+    series_one_stub.pop()
+
+    # Display the list.
+    print(series_one_stub)
+
+    # Ask the user for a fruit to delete, find it and delete it.
+    delete_fruit = input("enter a fruit to remove. CaSe SeNsItIvE:\n>")
+    if delete_fruit in series_one_stub:
+        series_one_stub.remove(delete_fruit)
+        print(series_one_stub)
+    else:
+        print("Try again, El Jefe!")
+        series_two()
+
+    # (Bonus: Multiply the list times two. Keep asking until
+    # a match is found. Once found, delete all occurrences.)
+    doubled_list = []
+    for fruit in series_one_stub:
+        doubled_list.append(fruit * 2)
+    processable = True
+    while processable:
+        delete_fruit = input("enter a times two fruit item to remove."
+                             " CaSe SeNsItIvE:\n>")
+        if delete_fruit in doubled_list:
+            doubled_list = list(filter(lambda a: a != delete_fruit, doubled_list))
```

Comment:
It's awesome that you're trying out some more advanced features of Python but I think you could achieve the same thing here with a simple while loop and the `list.remove()` method

Reply:
Indeed.  Sometimes I dont like lots of lines under an IF to make the blocks more readable for what actions are being taken at the cost of more complex and shorter logic.